File: springboot/src/main/java/com/orange/controller/system/GroupController.java
package com.orange.controller.system;

import com.orange.domain.common.Response;
import com.orange.domain.system.group.req.*;
import com.orange.entity.Group;
import com.orange.service.system.GroupService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.Parameter;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Tag(name = "组织管理", description = "组织管理相关接口")
@RestController
@RequestMapping("/api/system/groups")
@CrossOrigin
public class GroupController {
    @Autowired
    private GroupService groupService;

    @Operation(summary = "查询组织列表", description = "根据查询条件分页返回组织列表")
    @PostMapping
    @PreAuthorize("hasAuthority('menu#/system/group')")
    public Response<List<Group>> queryGroups(@RequestBody QueryGroupReq req) {
        List<Group> pageData = groupService.queryGroups(req);
        return Response.ok(pageData);
    }

    @Operation(summary = "删除组织", description = "根据组织ID删除组织")
    @DeleteMapping
    @PreAuthorize("hasAuthority('menu#/system/group:delete')")
    public Response<String> deleteGroups(@RequestBody DeleteGroupReq req) {
        groupService.deleteGroups(req.getIds());
        return Response.ok("操作成功");
    }

    @Operation(summary = "移动组织", description = "根据移动参数移动组织")
    @PostMapping("/move")
    @PreAuthorize("hasAuthority('menu#/system/group:edit')")
    public Response<String> moveGroup(@RequestBody MoveGroupReq req) {
        groupService.moveGroup(req);
        return Response.ok("操作成功");
    }

    @Operation(summary = "编辑组织", description = "根据编辑参数编辑组织")
    @PostMapping("/edit")
    @PreAuthorize("hasAuthority('menu#/system/group:edit')")
    public Response<String> editGroup(@RequestBody EditGroupReq req) {
        groupService.editGroup(req);
        return Response.ok("操作成功");
    }

    @Operation(summary = "查询组织详情", description = "根据组织ID查询组织详情")
    @GetMapping("/detail/{groupId}")
    @PreAuthorize("hasAuthority('menu#/system/group')")
    public Response<Group> queryGroup(
            @Parameter(description = "组织ID",example = "1")
            @PathVariable("groupId") String groupId) {
        Group group = groupService.queryGroup(groupId);
        return Response.ok(group);
    }

    @Operation(summary = "添加子组织", description = "添加子组织")
    @PostMapping("/addChild")
    @PreAuthorize("hasAuthority('menu#/system/group:add')")
    public Response<Group> addChild(@RequestBody @Valid AddChildGroupReq req) {
        Group group = groupService.addChild(req);
        return Response.ok(group);
    }

    @Operation(summary = "添加兄弟组织", description = "添加兄弟组织")
    @PostMapping("/addBrother")
    @PreAuthorize("hasAuthority('menu#/system/group:add')")
    public Response<Group> addBrother(@RequestBody @Valid AddBrotherGroupReq req) {
        Group group = groupService.addBrother(req);
        return Response.ok(group);
    }

    @Operation(summary = "查询所有组织", description = "查询所有组织")
    @GetMapping
    @PreAuthorize("hasAuthority('menu#/system/group')")
    public Response<List<Group>> queryGroups() {
        List<Group> groups = groupService.queryGroups();
        return Response.ok(groups);
    }

}

File: nestjs/src/system/group/group.controller.ts
import { Body, Controller, Delete, Get, Param, Post } from '@nestjs/common';
import { ResponseDto } from '../../common/dto/response.dto';
import { AddBrotherGroupDto } from './dto/add-brother-group.dto';
import { AddChildGroupDto } from './dto/add-child-group.dto';
import { DeleteGroupDto } from './dto/delete-group.dto';
import { EditGroupDto } from './dto/edit-group.dto';
import { MoveGroupDto } from './dto/move-group.dto';
import { QueryGroupDto } from './dto/query-group.dto';
import { GroupEntity } from './entities/group.entity';
import { GroupService } from './group.service';

@Controller('api/system/groups')
export class GroupController {
  constructor(private readonly groupService: GroupService) {}

  @Post()
  async queryGroups(
    @Body() query: QueryGroupDto,
  ): Promise<ResponseDto<GroupEntity[]>> {
    const list = await this.groupService.queryGroups(query);
    return ResponseDto.ok(list);
  }

  @Delete()
  async deleteGroups(
    @Body() dto: DeleteGroupDto,
  ): Promise<ResponseDto<string>> {
    await this.groupService.deleteGroups(dto.ids);
    return ResponseDto.ok('操作成功');
  }

  @Post('move')
  async moveGroup(@Body() dto: MoveGroupDto): Promise<ResponseDto<string>> {
    await this.groupService.moveGroup(dto);
    return ResponseDto.ok('操作成功');
  }

  @Post('edit')
  async editGroup(
    @Body() dto: EditGroupDto,
  ): Promise<ResponseDto<GroupEntity>> {
    const item = await this.groupService.editGroup(dto);
    return ResponseDto.ok(item);
  }

  @Get('detail/:groupId')
  async queryGroup(
    @Param('groupId') groupId: string,
  ): Promise<ResponseDto<GroupEntity>> {
    const item = await this.groupService.queryGroup(groupId);
    return ResponseDto.ok(item);
  }

  @Post('addChild')
  async addChild(
    @Body() dto: AddChildGroupDto,
  ): Promise<ResponseDto<GroupEntity>> {
    const item = await this.groupService.addChild(dto);
    return ResponseDto.ok(item);
  }

  @Post('addBrother')
  async addBrother(
    @Body() dto: AddBrotherGroupDto,
  ): Promise<ResponseDto<GroupEntity>> {
    const item = await this.groupService.addBrother(dto);
    return ResponseDto.ok(item);
  }

  @Get()
  async queryAllGroups(): Promise<ResponseDto<GroupEntity[]>> {
    const list = await this.groupService.queryGroups();
    return ResponseDto.ok(list);
  }
}
